index.html

```diff
@@ -1447,24 +1431,44 @@ <h2>The <dfn>NFCRecordType</dfn> enum</h2>
       Promise&lt;void&gt; push(NFCMessageSource message, optional NFCPushOptions options);
     };
 
-    [SecureContext]
-    interface NFC {
-      Promise&lt;void&gt; watch(MessageCallback callback, optional NFCWatchOptions options);
+    [Constructor(), SecureContext, Exposed=Window]
+    interface NFCReader : EventTarget {
+      Promise&lt;void&gt; watch(optional NFCWatchOptions options);
```

Comment:
Bikeshedding: Should we have a watch method, or should these be arguments for the constructor?

Reply:
To answer my own stupid question: It's because watch returns a promise when the initialization is complete.

To continue bikeshedding: Could this still take the options as an argument, and have a `readonly attribute Promise<void> loaded` to represent the return value of `watch`?

Reply:
I guess if you want to match the generic sensor API, there would just be `onerror` and `onactivate` events for this purpose. Seems like another decent option.

Reply:
Yes, these are follow up things, I also need the NFCReader ctor to take the options dict

Reply:
I don't think we need onactivate as we don't have start/stop (@reillyeon wanted to avoid that)